File: server/app/schemas/admin/analytics.py
from pydantic import BaseModel, ConfigDict
from typing import List

class KpiCard(BaseModel):
    title: str
    value: str
    change: str

class TopProduct(BaseModel):
    name: str
    value: int

class DeliveryStatusChart(BaseModel):
    on_time: int
    delayed: int

class OrderStatusBreakdownItem(BaseModel):
    status: str
    value: int

class AnalyticsSummary(BaseModel):
    kpi_cards: List[KpiCard]
    top_selling_products: List[TopProduct]
    delivery_status: DeliveryStatusChart
    order_status_breakdown: List[OrderStatusBreakdownItem]
    model_config = ConfigDict(from_attributes=True)

File: admin/src/types/analytics.ts
/**
 * Represents a single KPI card on the dashboard.
 */
export interface KpiCard {
  title: string;
  value: string;
  change?: string;
}

/**
 * Represents a top-selling product in analytics.
 */
export interface TopProduct {
  name: string;
  value: number;
}

/**
 * Data structure for delivery performance metrics.
 */
export interface DeliveryStatusChart {
  on_time: number;
  delayed: number;
}

/**
 * Item for the order status distribution chart.
 */
export interface OrderStatusBreakdownItem {
  status: string;
  value: number;
}

/**
 * Main summary object for the dashboard analytics.
 */
export interface AnalyticsSummary {
  kpi_cards: KpiCard[];
  top_selling_products: TopProduct[];
  delivery_status: DeliveryStatusChart;
  order_status_breakdown: OrderStatusBreakdownItem[];
}

/**
 * Represents a single revenue record for time-series charts.
 */
export interface RevenueDataPoint {
  date: string;
  revenue: number;
}

/**
 * Represents a product with inventory below the threshold.
 */
export interface LowStockProduct {
  name: string;
  stock_quantity: number;
}
